bodhi/server/models/models.py

```diff
@@ -789,7 +789,10 @@ def edit(cls, request, data):
 
                 # Expire any associated buildroot override
                 if b.override:
-                    b.override.expire()
+                    # Untag buildroot override only if the update reaches stable state or
+                    # the update reaches expired date
+                    if up.status == UpdateStatus.stable or datetime.utcnow() >= b.override.expired_date:
```

Comment:
Rather than having nested ifs, you can express the above as two ifs with an and statement:

```
if b.override and (up.status == UpdateStatus.stable or datetime.utcnow() >= b.override.expired_date):
```